modules/conflict_resolver/jsx/react.resolved_conflicts.js

```diff
@@ -0,0 +1,289 @@
+    };
+    this.formatColumn = this.formatColumn.bind(this);
+  }
+
+  updateFilterState(filter) {
+    this.setState({filter});
+  }
+
+  formatColumn(column, cell, rowData, rowHeaders) {
+    if (loris.hiddenHeaders.indexOf(column) > -1) {
+      return null;
+    }
+    // Create the mapping between rowHeaders and rowData in a row object.
+    let row = {};
+
+    rowHeaders.forEach(function(header, index) {
+      row[header] = rowData[index];
+    }, this);
+
+    if (column === 'Correct Answer') {
+      let correctAnswer = '';
+      let newValue = row['New Value'];
+      let oldValue1 = row['Correct Answer'];
+      let oldValue2 = row.OldValue2;
+
+      if (newValue === '1' && oldValue1 !== null) {
+        correctAnswer = oldValue1;
+      }
+
+      if (newValue === '2' && oldValue2 !== null) {
+        correctAnswer = oldValue2;
+      }
+
+      return <td>{correctAnswer}</td>;
+    }
+
+    return <td>{cell}</td>;
+  }
+
+  // Render the HTML
+  render() {
+    return (
+      <TabPane Title="" {...this.props}>
+        <RDynamicDataTable
+          DataURL={this.props.url.data.resolved}
+          Data={this.state.Data.Data}
+          Headers={this.state.Data.Headers}
+          Filter={this.state.filter}
+          getFormattedCell={this.formatColumn}
+        />
+      </TabPane>
+    );
+  }
+}
+ResolvedConflictsPane.propTypes = {
+  url: React.PropTypes.object.isRequired,
+  data: React.PropTypes.object.isRequired
+};
+ResolvedConflictsPane.defaultProps = {
+  module: "",
+  url: {
+    base: "",
+    data: {
+      unresolved: "",
+      resolved: ""
+    }
+  },
+  data: {}
+};
+
+/**
+ * This is the React class for the conflict resolver
+ */
+class ConflictsResolvedApp extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    loris.hiddenHeaders = [
+      "New Value",
+      "Old Value2",
+      "CenterID",
+      "Site"
+    ];
+
+    this.state = {
+      isLoaded: false,
+      filter: {},
+      form: {}
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   */
+  fetchData() {
+    $.ajax(this.props.url.data.resolved, {
+      method: "GET",
+      dataType: "json",
+      success: function(data) {
+        this.setState({
+          Data: data,
+          isLoaded: true
+        });
+      }.bind(this),
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  }
+
+  updateFilter(filter) {
+    this.setState({filter});
+    if (this.child !== undefined) {
+      this.child.updateFilterState(filter);
```

Comment:
I still don't think this is the best way to go about updating filter data. A change in the filter form should trigger a state update in the parent component which is then passed down as props to the child component -- not forcing its child to update its own state. We should be attempting to centralise state at the highest point of the component hierarchy, and pass that state down as necessary to children via props.